### PR TITLE
refactor: ImmutableERC1155 threat model

### PR DESCRIPTION
**Pull request by allan-almeida-imtbl**
Update threat model for `ImmutableERC1155` contract:
- Contract inheritance diagram & descriptions
- Document externally visible functions & access control
- Various formatting

### CODE REVIEW

File: audits/token/202312-threat-model-preset-erc1155.md [hunk 1]
```diff
@@ -1,28 +1,41 @@
 ## Introduction
-This document is a thread model for the preset erc1155 token contracts built by Immutable.
 
-This document encompasses information for all contracts under the [token](../contracts/token/erc1155) directory
+This document is a thread model for the preset ERC1155 token contracts built by Immutable.
+
+Contracts covered under this model include:
+
+- [ImmutableERC1155](../../contracts/token/erc1155/preset/ImmutableERC1155.sol)
```
Comment (drinkcoffee) [inline audits/token/202312-threat-model-preset-erc1155.md:7]: I would have this link to a specific githash. Given you are using the diagram from the README, it is conceivable that the diagram may change.

File: audits/token/202312-threat-model-preset-erc1155.md [hunk 2]
```diff
@@ -31,12 +44,79 @@ Permit is added to allow for Gasless transactions from the token owners.
...
+- The keys associated with the `DEFAULT_ADMIN_ROLE` should be operated by secure measures, for example a multi-signature wallet such that an attacker would need to compromise multiple signers simultaneously, or a securely stored hardware wallet.
+
+### Externally Visible Functions
+
+An attacker could formulate an attack in which they send one or more transactions that execute one or more of these functions.
+
+Functions that _change_ state:
+| Name | Function Selector | Access Control |
+| ------------------------------------------------------------- | ----------------- | --------------------- |
+| burn(address,uint256,uint256) | f5298aca | None - permisionless |
```
Comment (drinkcoffee) [inline audits/token/202312-threat-model-preset-erc1155.md:72]: I would expect the access control is that only the token owner can call burn.

Comment (allan-almeida-imtbl) [reply audits/token/202312-threat-model-preset-erc1155.md]: thanks, have addressed all state changing functions that previously mentioned being permissionless

**Review by ipekt — DISMISSED** [review]

File: audits/token/202312-threat-model-preset-erc1155.md (hunk 1 above)
Comment (allan-almeida-imtbl) [reply audits/token/202312-threat-model-preset-erc1155.md]: Have updated this link and also links for underlying contracts, to keep them all in sync: https://github.com/immutable/contracts/pull/229/commits/cef41759081a0a4a95587d0e4ccf4dab72e4777a

**Review by drinkcoffee — APPROVED** [review]

**Review by ipekt — APPROVED** [review]

**Pull request merged**